weave

```diff
@@ -238,14 +239,17 @@ elif [ "$1" = "run" -a -z "$IS_LOCAL" ] ; then
     # because we are passing arbitrary arguments on to docker run, and
     # we can't rely on our baked-in docker to support those arguments.
     shift 1
+    ATTACH_ARGS="--or-die"
     if [ "$1" != "--without-dns" ] ; then
         DNS_ARGS=$(exec_remote dns-args "$@" || true)
+        DNS_EXTRA_HOSTS_ARGS=$(exec_remote dns-extra-hosts "$@" || true)
```

Comment:
I know, but I honestly cannot think of a better way of doing it... I'm open to suggestions :confused: